crates/bevy_ecs/src/change_detection.rs

```diff
@@ -381,6 +381,18 @@ macro_rules! impl_methods {
                     ticks: self.ticks,
                 }
             }
+
+            /// Allows you access to the dereferenced value of this pointer without immediately
+            /// triggering change detection.
+            pub fn as_deref_mut(self)  -> Mut<'a, <$target as Deref>::Target>
```

Comment:
```suggestion
            pub fn as_deref_mut(&mut self)  -> Mut<'_, <$target as Deref>::Target>
```

Per convention, methods called `as` shouldn't consume self.

Reply:
I've tried this, but it fails because Mut contains TicksMut, which contains mutable references and cannot be copied (because it needs to still detect changes on the original), so they must be moved. If you know a way around this that would be great, otherwise may I should just change the method name to `to_deref_mut()`?

Reply:
Can you use `Mut::reborrow`, or do what it's doing?